### PR TITLE
[User Provider] Add a user provider for YAMCS

### LINKED ISSUE #92: Add user provider to expose logged in user
Yamcs provides information about the logged in user through the `/api/user` endpoint. It will return data that looks something like this:
```
{
  "superuser": false,
  "creationTime": "2021-12-16T16:41:15.103Z",
  "lastLoginTime": "2021-12-16T18:59:30.635Z",
  "active": true,
  "name": "user",
  "confirmationTime": "2021-12-16T16:41:15.103Z"
}
```

This can be used to define a user provider for the new User API that returns the logged in user's name.

### PR DESCRIPTION
**Pull request by jvigliotta**
Pulls information for logged in user from `api/user` and exposes the username (as well as other information).

closes #92

### CODE REVIEW

File: src/providers/user/createYamcsUser.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+export default function createExampleUser(UserClass) {
```
Comment (akhenry) [inline src/providers/user/createYamcsUser.js:23]: `createYamcsUser` maybe? Copy paste issue I think?

File: src/providers/user/user-provider.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        this.loggedIn = false;
+
+        this.YamcsUser = createYamcsUser(openmct.user.User);
+    }
+
+    isLoggedIn() {
+        return this.loggedIn;
+    }
+
+    isActive() {
```
Comment (akhenry) [inline src/providers/user/user-provider.js:39]: Let's leave this out for now, it's Yamcs specific so we wouldn't want to build any dependencies on it in our code code.

File: src/providers/user/user-provider.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        return this.user.isActive();
+    }
+
+    isSuperuser() {
```
Comment (akhenry) [inline src/providers/user/user-provider.js:43]: As above, let's leave this out for now.

**Review by akhenry — CHANGES_REQUESTED** [review]
Looks great! Some suggestions inline to keep this super-generic for now.

File: src/providers/user/createYamcsUser.js (hunk 1 above)
Comment (jvigliotta) [reply src/providers/user/createYamcsUser.js:23]: yep!

**Review by akhenry — APPROVED** [review]

**Pull request merged**